src/transformers/models/opt/modeling_opt.py

```diff
@@ -696,32 +841,12 @@ def forward(
         if inputs_embeds is None:
             inputs_embeds = self.embed_tokens(input_ids)
 
-        batch_size, seq_length = input_shape
         past_key_values_length = past_key_values[0][0].shape[2] if past_key_values is not None else 0
-        # required mask seq length can be calculated via length of past
-        mask_seq_length = past_key_values_length + seq_length
 
+        causal_attention_mask, attention_mask = self._update_causal_mask(
```

Comment:
Maybe a last nit: Is it necessary to keep the original attention mask over here?

Reply:
the embedding layer takes the attention mask as input, there is another PR I have waiting for some feedback to change some of the logic there

https://github.com/huggingface/transformers/pull/33121#pullrequestreview-2283357050

Reply:
Alright, makes sense. Just showed my unfamiliarity with opt as a model ^^